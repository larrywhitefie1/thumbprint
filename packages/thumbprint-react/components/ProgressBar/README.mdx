---
package: '@thumbtack/thumbprint-react'
kit: progress-bar/index.mdx
platform: react
url: /components/progress-bar/react/
mdxType: componentApi
---

## Basic progress bar

```jsx
<ProgressBar value={70} />
```
