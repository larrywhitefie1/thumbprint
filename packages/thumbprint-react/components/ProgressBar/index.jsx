import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import ProgressBarBase from '../ProgressBarBase/index.jsx';
import styles from './index.module.scss';

const ProgressBar = ({ value, shouldShowLabel, labelAlign }) => (
    <div>
        <ProgressBarBase width={value} />

        {shouldShowLabel && (
            <div
                className={classnames({
                    [styles.label]: true,
                    [styles.labelCenter]: labelAlign === 'center',
                })}
            >
                {value}% complete
            </div>
        )}
    </div>
);

ProgressBar.propTypes = {
    /**
     * A number between 0 and 100 that indicates the progress made.
     */
    value: PropTypes.number,
    labelAlign: PropTypes.oneOf(['left', 'center']),
    shouldShowLabel: PropTypes.bool,
};

ProgressBar.defaultProps = {
    value: 0,
    labelAlign: 'left',
    shouldShowLabel: true,
};

export default ProgressBar;
