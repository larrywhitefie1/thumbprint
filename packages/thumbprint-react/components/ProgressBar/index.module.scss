@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';
@import '~@thumbtack/tp-ui-core-function/_index';

.label {
    font-size: $tp-font__body__3__size;
    margin-top: tp-spacer(1);
    text-align: left;
}

.labelCenter {
    text-align: center;
}
