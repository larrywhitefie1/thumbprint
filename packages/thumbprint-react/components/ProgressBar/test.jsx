import React from 'react';
import { shallow, mount } from 'enzyme';
import ProgressBar from './index';

test('passes progress to base component', () => {
    const wrapper = shallow(<ProgressBar value={33} />);
    const width = wrapper.find('ProgressBarBase').prop('width');
    expect(width).toBe(33);
    expect(wrapper).toMatchSnapshot();
});

test('shows label when `shouldShowLabel` is true', () => {
    const wrapper = shallow(<ProgressBar shouldShowLabel />);
    const label = wrapper.find('.label');
    expect(label).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});

test('shows label when `shouldShowLabel` is false', () => {
    const wrapper = shallow(<ProgressBar shouldShowLabel={false} />);
    const label = wrapper.find('.label');
    expect(label).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
});

test('aligns label text with `labelAlign`', () => {
    const wrapperA = mount(<ProgressBar labelAlign="left" />);
    expect(wrapperA).toMatchSnapshot();

    const wrapperB = mount(<ProgressBar labelAlign="center" />);
    expect(wrapperB).toMatchSnapshot();
});
