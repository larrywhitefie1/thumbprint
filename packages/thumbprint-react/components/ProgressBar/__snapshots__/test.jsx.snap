// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aligns label text with \`labelAlign\` 1`] = `
<ProgressBar
  labelAlign="left"
  shouldShowLabel={true}
  value={0}
>
  <div>
    <ProgressBarBase
      width={0}
    >
      <div
        className="outer"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={0}
          className="inner"
          role="progressbar"
          style={
            Object {
              "transform": "translateX(-100%)",
            }
          }
        />
      </div>
    </ProgressBarBase>
    <div
      className="label"
    >
      0
      % complete
    </div>
  </div>
</ProgressBar>
`;

exports[`aligns label text with \`labelAlign\` 2`] = `
<ProgressBar
  labelAlign="center"
  shouldShowLabel={true}
  value={0}
>
  <div>
    <ProgressBarBase
      width={0}
    >
      <div
        className="outer"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={0}
          className="inner"
          role="progressbar"
          style={
            Object {
              "transform": "translateX(-100%)",
            }
          }
        />
      </div>
    </ProgressBarBase>
    <div
      className="label labelCenter"
    >
      0
      % complete
    </div>
  </div>
</ProgressBar>
`;

exports[`passes progress to base component 1`] = `
<div>
  <ProgressBarBase
    width={33}
  />
  <div
    className="label"
  >
    33
    % complete
  </div>
</div>
`;

exports[`shows label when \`shouldShowLabel\` is false 1`] = `
<div>
  <ProgressBarBase
    width={0}
  />
</div>
`;

exports[`shows label when \`shouldShowLabel\` is true 1`] = `
<div>
  <ProgressBarBase
    width={0}
  />
  <div
    className="label"
  >
    0
    % complete
  </div>
</div>
`;
